Automatically show attributes in Variations

octaedro opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->
This PR adds the code to automatically show attributes with "Used for variations" checked, in Variations.

Fixes partially [issue 33551](https://github.com/woocommerce/woocommerce/issues/33551).

**Acceptance criteria:**

- [x] If there are attributes with "Used for variations" checked: automatically show them in Variations, rather than force people to add them manually

---

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested (including pre-conditions, configuration, steps to take and expected results). It may help to write your instructions using pseudocode -- as if you're telling a computer how to execute the test. -->

1. Go to `Products` > `Add New` > `Product data` > `Attributes`
2. Add a custom attribute by clicking on `Add custom attribute`. Also add a few `values`.
3. Press `Save attributes`.
4. Press `Variations`.
5. You'll see an alert saying the number of variations that will be added.

![Screenshot 2022-12-01 at 16 45 47](https://user-images.githubusercontent.com/1314156/205145717-9e95ba66-1284-4436-9c8e-48b221e22d1e.png)

6. Verify that the attributes have been added correctly. 

![Screenshot 2022-12-01 at 16 48 13](https://user-images.githubusercontent.com/1314156/205145920-c3bf4cba-19be-4d67-ad26-c481888a46f5.png)

<!-- End testing instructions -->

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

octaedro committed:
Automatically show attributes in Variations

# Conflicts:
#	plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js

octaedro committed:
Add changelog

joshuatf commented on plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js:
```diff
@@ -751,7 +751,17 @@ jQuery( function ( $ ) {
+				var isUsedForVariations = $( 'input#used-for-variation' ).is(
+					':checked'
+				);
+				$( 'select.wc-attribute-search' ).data(
```
I think this selector is only available if there are no attributes globally already available.

This could also be `attribute_taxonomy`.

https://github.com/woocommerce/woocommerce/blob/98a82c018cff47419dc7811bcaf6d685721da3f5/plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-attributes.php#L40

octaedro replied:
Nice catch! I fixed that in the commit [88978ef](https://github.com/woocommerce/woocommerce/pull/35807/commits/88978eff835f76831344a60e9288eb4144b07382)

joshuatf commented on plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js:
```diff
@@ -751,7 +751,17 @@ jQuery( function ( $ ) {
+					':checked'
+				);
+				$( 'select.wc-attribute-search' ).data(
+					'add-attribute-used-for-variations',
```
Nitpicky, but could we make this `is-used-for-variations` which defines state for this item instead of an intent?

octaedro replied:
Good idea; I changed it in the commit [aa0f99b](https://github.com/woocommerce/woocommerce/pull/35807/commits/aa0f99b2a88faed0d1580d561eaf2e19c679ab93)

joshuatf reviewed (commented):
Nice clean solution, Fernando 😄 

Left a couple minor comments.  Let me know if you have any questions.

octaedro committed:
Use `attribute_taxonomy` instead of `wc-attribute-search`.

# Conflicts:
#	plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js

octaedro committed:
Rename `add-attribute-used-for-variations`

octaedro commented:
Thank you @joshuatf for your review!
I addressed the changes you suggested. Could you take another look at this PR?

joshuatf reviewed (approved):
Testing well and code looks good!  Thanks for the changes, Fernando 💯
